src/components/__tests__/form.test.tsx

```diff
@@ -1,26 +1,85 @@
 // Copyright (c) 2018 Ultimaker B.V.
-import 'jest';
-import 'jsdom-global/register';
 import * as React from 'react';
-import { shallow } from 'enzyme';
+import { shallow, mount } from 'enzyme';
 
 // component
-import Form from '../form';
+import { Form } from '../form';
+import { InputField } from '../input_field';
+import { Button } from '../button';
+
+// mocks
+import { mockClickEvent } from '../../__mocks__/clickMock'
 
 describe('The Form component', () => {
-  let props;
-  let wrapper;
-
-  beforeEach(() => {
-    props = {
-      onSubmitHandler: jest.fn(),
-    };
-    wrapper = shallow(<Form {...props} />);
-  });
-
-  it('should render', () => {
-    expect(wrapper).toMatchSnapshot();
-    expect(props.onSubmitHandler).not.toHaveBeenCalled();
-  });
+    let props;
+    let wrapper;
+
+    beforeEach(() => {
+        props = {
+            onSubmitHandler: jest.fn(),
+            primaryBtnText: 'primaryBtnText'
+        };
+        wrapper = shallow(<Form {...props} />);
+    });
+
+    it('should render', () => {
+        expect(wrapper).toMatchSnapshot();
+        expect(props.onSubmitHandler).not.toHaveBeenCalled();
+    });
+
+    it('should call onSubmitHandler when form is submitted', () => {
+        wrapper.simulate('submit', mockClickEvent);
+        expect(props.onSubmitHandler).toHaveBeenCalled();
+        expect(wrapper.state().submitted).toEqual(true);
```

Comment:
I propose to use state() as we agreed in the case of props().
state('key') instead of state().key